gradio/routes.py

```diff
@@ -584,6 +584,55 @@ async def reset_iterator(body: ResetBody):
                 await app.get_blocks()._queue.clean_events(event_id=body.event_id)
             return {"success": True}
 
+        @app.get("/heartbeat/{session_hash}")
+        def heartbeat(
+            session_hash: str,
+            request: fastapi.Request,
+            background_tasks: BackgroundTasks,
+            username: str = Depends(get_current_user),
+        ):
+            """Clients make a persistent connection to this endpoint to keep the session alive.
+            When the client disconnects, the session state is deleted.
+            It is robust to internet cutting out, etc.
+            """
+
+            async def iterator():
+                while True:
```

Comment:
I did the experiment where I killed my network connection and the state did not get deleted when I reconnected.

![network_kill](https://github.com/gradio-app/gradio/assets/41651716/667f97bf-90eb-4c14-910d-a194d079469c)

Reply:
Nice! How's that possible though? An asyncio.CancelledError should be raised when the network disconnects right?

Reply:
Maybe chrome keeps the connections alive but doesn't allow new requests to go through?

Reply:
Yes that's what I was thinking as well